google/ggrc-core
# GGRC-4734 GC/GR user can map any object to Audit where he is Audit Captain

# Issue description

Application do not validate permissions for Snapshotable objects

# Steps to test the changes

Steps to reproduce:
1. Create Audit and Control
2. Assign Global Creator as Audit Captain for Audit from step#1
3. Log as Global Creator and open Audit page
4. Send POST request on /api/relationships to create relationship between Control and Audit from step#1
5. Reload page and open Controls tab
Actual Result: If user is Audit Captain in Audit and doesn't have any rights on Control (or any other snapshotable object) he can map this object to Audit and see it through the api.
Expected Result: Audit Captain should not be able to map object to Audit and see it if he doesn't have any rights on this object
This issue is reproduced through the api only with POST request on /api/relationships

# Solution description

Add validation that we have access to update child object of Snapshot

# Sanity checklist




# PR Review checklist
